qiime2/sdk/result.py

```diff
@@ -282,6 +286,205 @@ def result(self):
+
+        Parameters
+        ----------
+        name : str\n
+            The name of the Annotation to retrieve.
+
+        Returns
+        -------
+        Annotation : obj\n
+            The Annotation object associated with the provided name.
+
+        Raises
+        ------
+        ValueError\n
+            If no Annotation with the provided name is found.
+
+        """
+        self._validate_annotation_support()
+
+        for annotation in self._annotations:
+            if annotation.name == name:
+                return annotation
+
+        raise ValueError(f'No Annotation with name: "{name}" was found.')
+
+    # TODO: add support to filter by type
+    # once additional annotation types are added in 7.1
+    def iter_annotations(self):
+        """Constructs an iterable containing all Annotations associated with
+        the Result object.
+
+        Raises
+        ------
+        ValueError\n
+            If no Annotations are found associated with the Result.
+
+        """
+        self._validate_annotation_support()
+
+        if self._annotations:
+            return iter(self._annotations)
+
+        raise ValueError('No Annotations found.')
+
+    def remove_annotation(self, name):
+        """
+        Remove an Annotation given by `name` from the Result object.
+
+        Parameters
+        ----------
+        name : str\n
+            The name of the Annotation to be removed.
+
+        Raises
+        ------
+        ValueError\n
+            1. If there are no Annotations associated with the Result object.
+            2. If no Annotation with the specified name is found.
+            3. If the corresponding annotation directory cannot be located.
+
+        """
+        self._validate_annotation_support()
+        # First check that annotations dir exists for the Result
+        annotations_dir = self._archiver.annotations_dir
+        if not os.path.exists(annotations_dir):
+            raise ValueError('No existing annotations found.')
+
+        annotation_to_remove = None
+        for annotation in self._annotations:
+            if annotation.name == name:
+                annotation_to_remove = annotation
+                break
+
+        # Guard against provided Annotation name not found on Result object
+        if annotation_to_remove is None:
+            raise ValueError(f'No Annotation found with name: "{name}"')
+
+        # Check for corresponding Annotation entry on disk
+        annotation_disk_dir = None
+
+        for entry in os.listdir(annotations_dir):
+            subdir = os.path.join(annotations_dir, entry)
+            meta_filepath = os.path.join(subdir, 'metadata.yaml')
+
+            if os.path.isfile(meta_filepath):
+                with open(meta_filepath, 'r') as fh:
+                    meta = yaml.safe_load(fh)
+
+                if meta.get('name') == name:
+                    annotation_disk_dir = subdir
+                    break
```

Comment:
For this, we have already loaded the annotation and as such we should have its uuid right? So this should look something like 

`shutil.rmtree(os.path.join(annotations_dir, str(annotation_to_remove.uuid)))`

Reply:
nice, yeah you're right Anthony. wrote this one too late in the day to be thinking clearly lol